src/proto/client/conn.rs

```diff
@@ -262,6 +262,12 @@ fn handle_packet(packet: ReceivedPacket, state: &Arc<Mutex<ClientState>>) -> cra
                 state.lock().unwrap().clear_entries();
             }
         }
+        ReceivedPacket::RpcResponse(rpc) => {
+            let mut state = state.lock().unwrap();
+            if let Some(callback) = state.rpc_callbacks.remove(&rpc.unique_id) {
+                callback(rpc.result);
```

Comment:
Evaluate the callback asynchronously, we don't want user code accidentally stalling the network loop

Reply:
fixed